File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        int l = 0;
        int r = n - 1;
        int idx = n - 1;
        
        while (l <= r)
            if (Math.abs(nums[l]) > Math.abs(nums[r]))
                result[idx--] = nums[l] * nums[l++];
            else 
                result[idx--] = nums[r] * nums[r--];
            
        return result;
    }
}

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        int[] squares = new int[nums.Length];
        int left = 0, right = nums.Length - 1;
        
        for (int i = nums.Length - 1; i >= 0; i--) {
            if (Math.Abs(nums[left]) > Math.Abs(nums[right]))
                squares[i] = nums[left] * nums[left++];
            else squares[i] = nums[right] * nums[right--];
        }
        
        return squares;
    }
}